service-workers/service-worker/fetch-event.https.html

```diff
@@ -605,5 +608,131 @@
     frame.remove();
     await service_worker_unregister_and_done(t, scope);
   }, 'Service Worker responds to fetch event with the correct keepalive value');
+
+promise_test(async (t) => {
+    const scope = 'resources/simple.html?isReloadNavigation';
+    let frame;
+
+    try {
+      const reg = await service_worker_unregister_and_register(t, worker, scope);
+      await wait_for_state(t, reg.installing, 'activated');
+      frame = await with_iframe(scope);
+      assert_equals(frame.contentDocument.body.textContent,
+                    'method = GET, isReloadNavigation = false');
+      await new Promise((resolve) => {
+        frame.addEventListener('load', resolve);
+        frame.contentWindow.location.reload();
+      });
+      assert_equals(frame.contentDocument.body.textContent,
+                    'method = GET, isReloadNavigation = true');
+    } finally {
+      if (frame) {
+        frame.remove();
+      }
+      await service_worker_unregister(t, scope);
+    }
+  }, 'Service Worker responds to fetch event with the correct isReloadNavigation value (location.reload())');
+
+promise_test(async (t) => {
+    const scope = 'resources/simple.html?isReloadNavigation';
+    let frame;
+    try {
+      const reg = await service_worker_unregister_and_register(t, worker, scope);
+      await wait_for_state(t, reg.installing, 'activated');
+      frame = await with_iframe(scope);
+      assert_equals(frame.contentDocument.body.textContent,
+                    'method = GET, isReloadNavigation = false');
+      await new Promise((resolve) => {
+        frame.addEventListener('load', resolve);
+        frame.contentWindow.history.go(0);
+      });
+      assert_equals(frame.contentDocument.body.textContent,
+                    'method = GET, isReloadNavigation = true');
+    } finally {
+      if (frame) {
+        frame.remove();
+      }
+      await service_worker_unregister(t, scope);
+    }
+  }, 'Service Worker responds to fetch event with the correct isReloadNavigation value (historg.go(0))');
```

Comment:
I don't think the "Service Worker responds..." part is really needed but I know existing tests have this. Maybe just call it:
'FetchEvent#request.isReloadNavigation is true for history.go(0)"

Reply:
Done.